File: Solutions/2239. Find Closest Number to Zero/2239. Find Closest Number to Zero.cpp
class Solution {
public:
    int findClosestNumber(vector<int>& nums) {
        return *min_element(begin(nums),end(nums),[&](int n1,int n2){
            return abs(n1)<abs(n2) or (abs(n1)==abs(n2) and n1>n2);
        });
    }
};

File: Solutions/2239. Find Closest Number to Zero/2239. Find Closest Number to Zero.go
func findClosestNumber(nums []int) int {
    res:=nums[0]
    for i:=1;i<len(nums);i+=1{
        if abs(nums[i])<abs(res){
            res=nums[i]
        }else if abs(nums[i])==abs(res) && nums[i]>res{
            res=nums[i]
        }
    }
    return res
}

func abs(x int) int {
    if x<0{
        return -x
    }
    return x
}